scripts/code_generation/commit_info_gen/src/generate_commit_info.py

```diff
@@ -0,0 +1,67 @@
+import os
+import string
+from argparse import ArgumentParser
+import jinja2
+import git
+
+# validates data dict
+# data["hash"]: 7 char string with short commit hash
+# data["clean"]: "0" if dirty, "1" is clean
+def validateData(data):
+  validLength = len(data["hash"]) == 7
+  validHex = all([c in string.hexdigits for c in data["hash"]])
+  validClean = data["clean"] == "0" or data["clean"] == "1"
+  return validLength and validHex and validClean
+
+if __name__ == "__main__":
+
+  # set up path arg
+  parser = ArgumentParser()
+  parser.add_argument("--output-header", type=str, required=True)
+  parser.add_argument("--output-source", type=str, required=True)
+  args = parser.parse_args() 
+
+  # get template for rendering
+  module_dir = os.path.dirname(os.path.relpath(__file__))
+  template_dir = os.path.join(module_dir, "templates")
+  env = jinja2.Environment(loader=jinja2.FileSystemLoader(template_dir))
+  header_template = env.get_template("./App_CommitInfo.h.j2")
+  source_template = env.get_template("./App_CommitInfo.c.j2")
+
+  # data to expose to header
+  data = {}
+
+  try: 
+    # get commit info
+    repo = git.Repo(search_parent_directories=True)
+    commit = repo.head.commit
+    clean = not repo.is_dirty(untracked_files=True)
+
+    # short hash is the first 7 chars of the long hash
+    short_hash = commit.hexsha[0:7]
+
+    data["hash"] = short_hash
+    data["clean"] = "1" if clean else "0"
+
+    # validate
+    if not validateData(data):
+      raise Exception("Invalid data.")
+
+  except:
```

Comment:
i would suggest catching the specific error here we are expecting in order to allow other errors to stop the process

Reply:
Tried to make things a bit clearer in commit e55adf4249ba968141137c430e981a249d7c51a1.

Previously, one try catch would handle both GitPython and data validation. Now data validation is not in the try except, and the try except handles GitPython errors only.

Because commit_info_gen is non-critical, we want to stop it from blocking compilation as much as possible. To do this, if any error is released by GitPython, dummy data is returned instead, with a warning logged to the terminal. We don't care what type of error is thrown, all errors should be caught so the process can continue.